src/app/pages/questions/components/question/text-input/text-input.component.ts

```diff
@@ -21,22 +22,26 @@ export class TextInputComponent implements OnInit {
   customMonthNames = []
   datetimeValue: { [key: string]: any } = {}
 
+  constructor(
+    private localization: LocalizationService
+  ) {}
+
+
   ngOnInit() {
-    this.initMonthNames()
     this.showDatePicker = this.type.includes('date')
     this.showTimePicker = this.type.includes('time')
     this.showSeconds = this.type.includes('seconds')
     this.initFormats()
   }
 
-  initMonthNames() {
-    for (let x = 1; x <= 12; x++)
-      this.customMonthNames.push(moment(x, 'M').format('MMM'))
-  }
 
   initFormats() {
-    this.dateFormat = 'MMM DD YYYY'
-    this.timeFormat = this.showSeconds ? 'HH:mm:ss' : 'HH:mm'
+    const locale = this.localization.moment().localeData()
+    this.dateFormat = locale.longDateFormat('LL').replace('MMMM', 'MMM')
+    this.timeFormat = this.showSeconds ? locale.longDateFormat('LTS') : locale.longDateFormat('LT')
```

Comment:
Maybe we can keep the time format constant so that it will always use the 24 hour time? So that this can also be used for asking the duration of something in addition to asking the time?

Reply:
Hmm yeah unfortunately redcap doesn't have a dedicated duration field type, so you have to use a time field for asking durations. But I'm not sure how to do this in a better way. Some ways I can think of:
- a custom field in the definition indicating duration type, or just manually changing the `field_type` to `duration`
- just looking for the string `duration` in the `field_name` (e.g. like [here](https://github.com/RADAR-base/RADAR-REDCap-aRMT-Definitions/blob/2ad9a4d4d3e3fdb85e48d6184c3f00708a9e8b6e/questionnaires/seizure_diary/seizure_diary_armt.json#L24))
- A button on the page allowing a user to switch between duration and time input (defaulting to time)

Currently this would look like this, which is kind of unintuitive:
![Screenshot_1568120073](https://user-images.githubusercontent.com/5064873/64615556-111e2e80-d3db-11e9-9adc-9190411c5e00.png)

Reply:
Yes, I think one solution could be to add `duration` it to the `field_annotation` and parse it in the definitions utility script. This is similar to how we do this for other kinds of inputs which are not provided directly in Redcap (see https://github.com/RADAR-base/RADAR-REDCap-aRMT-Definitions/blob/d64b72c883449daf62c04c913806b1190516e482/utilities/index.js#L117)